calculation/gmhazard_calc/gmhazard_calc/directivity/directivity.py

```diff
@@ -157,40 +159,63 @@ def compute_fault_directivity(
         constants.EventType.from_rake(rake),
     )
 
-    # Creating the array to store all fdi values
-    fdi_array = np.zeros((n_hypo_data.nhypo, len(sites), len(periods)))
-    phired_array = np.zeros((n_hypo_data.nhypo, len(sites), len(periods)))
-    for index, planes in enumerate(hypo_planes):
-        # Gets the plane index of the hypocentre
-        plane_index = plane_ind[index]
+    if n_procs == 1:
+        fdi, phired = [], []
+        for ix, cur_planes in enumerate(hypo_planes):
+            cur_fdi, (cur_phi_red, cur_predictor_functions, cur_other) = _compute_directivity_effect(
+                lon_lat_depth,
+                cur_planes,
+                plane_ind[ix],
+                sites,
+                nominal_strike,
+                nominal_strike2,
+                mag,
+                rake,
+                periods,
+            )
 
-        fdi, (phi_red, predictor_functions, other) = _compute_directivity_effect(
-            lon_lat_depth,
-            planes,
-            plane_index,
-            sites,
-            nominal_strike,
-            nominal_strike2,
-            mag,
-            rake,
-            periods,
-        )
+            fdi.append(cur_fdi)
+            phired.append(cur_phi_red)
+    else:
+        # Compute directivity for all hypocentres
+        with mp.Pool(n_procs) as pool:
+            results = pool.starmap(
+                _compute_directivity_effect,
+                [
+                    (
+                        lon_lat_depth,
+                        cur_planes,
+                        plane_ind[ix],
+                        sites,
+                        nominal_strike,
+                        nominal_strike2,
+                        mag,
+                        rake,
+                        periods,
+                    )
+                    for (ix, cur_planes) in enumerate(hypo_planes)
+                ],
+            )
+
+        # Combine results
+        fdi = [cur_result[0] for cur_result in results]
+        phired = [cur_result[1][0] for cur_result in results]
```

Comment:
Just one question, could this be an alternative approach?
1. Declare `fdi`, `phired` before `if n_procs == 1:`
2. If not doing multiprocessing, nothing changes as it is already using `append`
3. But with multiprocessing, while combining results, instead of doing for loop on `cure_result` twice, do
```python
for cur_result in results:
    fdi.append(cur_result[0]
    phired.append(cur_result[1][0]
```

Reply:
I see what you are saying here, but there isn't really a difference compared to what I have. Looping over twice does not matter in this case imo, since the number if iterations is super small (<100)

Reply:
Fair point, just worried as the dataframes I've been playing with last few days, they were too big :joy: